### PR TITLE
new(tests): EOF - Tests from Fuzzing

### PR DESCRIPTION
**Pull request by shemnon**
## 🗒️ Description

A collection of fuzzing related updates to the test

* Activate functions validation tests, and move to 4750 dir
* Add terminal opcode to some tests so they have only one error
* Tests found via fuzzing coverage
  * subcontainer wrong size
  * subcontainer wrong eof version
  * EOFCREATE nonexistant container
  * Many CALLF test covered by inactive 4750 checks

## ✅ Checklist
- [ ] All: Set appropriate labels for the changes.
- [ ] All: Considered squashing commits to improve commit history.
- [ ] All: Added an entry to [CHANGELOG.md](/ethereum/execution-spec-tests/blob/main/docs/CHANGELOG.md).
- [ ] All: Considered updating the online docs in the [./docs/](/ethereum/execution-spec-tests/blob/main/docs/) directory.
- [ ] Tests: All converted JSON/YML tests from [ethereum/tests](/ethereum/tests) have been added to [converted-ethereum-tests.txt](/ethereum/execution-spec-tests/blob/main/converted-ethereum-tests.txt).
- [ ] Tests: A PR with removal of converted JSON/YML tests from [ethereum/tests](/ethereum/tests) have been opened.
- [ ] Tests: Included the type and version of evm t8n tool used to locally execute test cases:  e.g., ref with commit hash or geth 1.13.1-stable-3f40e65.
- [ ] Tests: Ran `mkdocs serve` locally and verified the auto-generated docs for new tests in the [Test Case Reference](https://ethereum.github.io/execution-spec-tests/main/tests/) are correctly formatted.

### CODE REVIEW

File: tests/prague/eip7692_eof_v1/eip7620_eof_create/test_subcontainer_validation.py [hunk 1]
```diff
@@ -471,6 +471,55 @@ def test_container_both_kinds_different_sub(eof_test: EOFTestFiller):
     )
 
 
+@pytest.mark.parametrize("version", [0, 255], ids=lambda x: x)
+def test_subcontainer_wrong_eof_version(
+    eof_test: EOFTestFiller,
+    version: int,
+):
+    """Test multiple kinds of subcontainer at the same level"""
```
Comment (marioevz) [inline tests/prague/eip7692_eof_v1/eip7620_eof_create/test_subcontainer_validation.py:479]:
```suggestion
    """Test a subcontainer with the incorrect EOF version"""
```

File: tests/prague/eip7692_eof_v1/eip7620_eof_create/test_subcontainer_validation.py [hunk 2]
```diff
@@ -471,6 +471,55 @@ def test_container_both_kinds_different_sub(eof_test: EOFTestFiller):
...
+
+
+@pytest.mark.parametrize("delta", [-1, 1], ids=["smaller", "larger"])
+@pytest.mark.parametrize("kind", [ContainerKind.RUNTIME, ContainerKind.INITCODE])
+def test_subcontainer_wrong_size(
+    eof_test: EOFTestFiller,
+    delta: int,
+    kind: ContainerKind,
+):
+    """Test multiple kinds of subcontainer at the same level"""
```
Comment (marioevz) [inline tests/prague/eip7692_eof_v1/eip7620_eof_create/test_subcontainer_validation.py:503]:
```suggestion
    """Test a subcontainer with the incorrect size in the parent's header"""
```

**Review by marioevz — APPROVED** [review]
LGTM, just a couple of small comments. Thanks!

File: tests/prague/eip7692_eof_v1/eip4200_relative_jumps/test_rjump.py [hunk 3]
```diff
@@ -696,3 +696,49 @@ def test_rjump_backwards_reference_only(
...
+        data=Container(
+            name="infinite_loop",
+            sections=[
+                Section.Code(
+                    code=Op.PUSH0
+                    + Op.RJUMPI[3]
+                    + Op.RJUMP[7]
+                    + Op.SSTORE(1, 0x2015)
+                    + Op.STOP
+                    + Op.RJUMP[-10]
```
[the remark this answers is not in the document]
Comment (shemnon) [reply tests/prague/eip7692_eof_v1/eip4200_relative_jumps/test_rjump.py:739]: I mis-read it.  The unique value appears to be a backwards RJUMP as a terminal

File: tests/prague/eip7692_eof_v1/eip7620_eof_create/test_subcontainer_validation.py (hunk 2 above)
Comment (shemnon) [reply tests/prague/eip7692_eof_v1/eip7620_eof_create/test_subcontainer_validation.py]: fixed

File: tests/prague/eip7692_eof_v1/eip7620_eof_create/test_subcontainer_validation.py (hunk 1 above)
Comment (shemnon) [reply tests/prague/eip7692_eof_v1/eip7620_eof_create/test_subcontainer_validation.py]: fixed

File: tests/prague/eip7692_eof_v1/eip3540_eof_v1/code_validation_function.py [hunk 4]
```diff
@@ -1,189 +0,0 @@
...
-            Section.Code(
-                code=Op.PUSH0 + Op.RETF,
-                code_inputs=1,
-                code_outputs=2,
-                max_stack_height=2,
-            ),
-        ],
-    ),
-    Container(
-        name="stack_height_equal_code_outputs_retf_zero_stop",
```
Comment (pdobacz) [inline tests/prague/eip7692_eof_v1/eip3540_eof_v1/code_validation_function.py:30]: this test is being removed, but looks legit, can you confirm removal is intended?

**Comment by pdobacz** [comment]
@shemnon If the new test cases cover cases reported in https://github.com/ipsilon/eof/issues/146, please tick them off as "covered in EEST" with a corresponding link maybe (best by editing the respective comments there). Thank you!

**Comment by marioevz** [comment]
Please feel free to merge once the comments by @pdobacz have been addressed and conflicts have been resolved.

**Comment by shemnon** [comment]
@marioevz - I added a few more tests from the fuzzing page.  Tried to reduce them as far as I could to their simplest state.

**Pull request merged**